sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -935,6 +935,104 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
     sql("DROP TABLE t1")
   }
 
+  test("LOAD DATA") {
+    sql(
+      """
+        |CREATE EXTERNAL TABLE non_part_table (time TIMESTAMP, id INT)
+        |ROW FORMAT DELIMITED
+        |FIELDS TERMINATED BY ','
+        |LINES TERMINATED BY '\n'
+      """.stripMargin)
+
+    val testData = TestHive.getHiveFile("data/files/issue-4077-data.txt").getCanonicalPath
+
+    // LOAD DATA INTO non-partitioned table can't specify partition
+    intercept[AnalysisException] {
+      sql(s"""LOAD DATA LOCAL INPATH "$testData" INTO TABLE non_part_table PARTITION(ds="1")""")
+    }
+
+    sql(s"""LOAD DATA LOCAL INPATH "$testData" INTO TABLE non_part_table""")
+    assert(sql("SELECT time FROM non_part_table LIMIT 2").collect()
```

Comment:
I think it's because we parse the LOAD DATA command in `HiveSqlParser`, so we have to be in hive context.